src/material/datepicker/_datepicker-legacy-compat.scss

```diff
@@ -6,6 +6,10 @@
     @include button-theme.theme($theme);
     @include icon-button-theme.theme($theme);
   }
+
+  .mat-datepicker-toggle {
```

Comment:
I ran this PR against google's internal tests and it looks like it caused some sizing issues in a few apps. I suspect its a specificity issue. Could you try adding an extra class of specificity to the datepicker toggle styles in legacy-button-compat. I think that should compensate for the extra specificity added here

Reply:
All the styles or just the ones added by this PR?

Reply:
No worries, I pushed a commit with the change I was imagining. Let me see if it helps with the tests

Reply:
Oh, a change was needed in the `legacy-button-compat` mixin? 🤔 Interesting.